component/admin/views/media/tmpl/default.php

```diff
@@ -98,16 +98,20 @@
         <div class="filterTool">
             <?php if ($showbuttons != 1): ?>
                 <div class="filterItem">
-                    <?php echo JText::_('COM_REDSHOP_MEDIA_TYPE') . ': ' . $this->lists['type']; ?>
+                    <div class="btn-wrapper input-append">
+                        <input type="text" name="filter_search" id="filter_search" value="<?php echo $this->filter_search; ?>"
+                               placeholder="<?php echo JText::_('JSEARCH_FILTER'); ?>"/>
+                        <button class="btn" onclick="document.adminForm.submit();"><?php echo JText::_(
+                                'COM_REDSHOP_SEARCH'
+                            ); ?></button>
+                        <input type="button" value="Reset" class="btn reset" onclick="this.form.getElementById('media_type').value='0';this.form.getElementById('filter_media_section').value='0';this.form.submit();"/>
```

Comment:
moving code inside `onclick` to a js function, adding reset for the keyword input